Product.cs

```diff
@@ -10,24 +11,28 @@ public class Product
     {
         public string Name { get; set; }
         public int UPC { get; set; }
-        public decimal BasicPrice { get; set; }
+        public static decimal BasicPrice { get; set; }
 
-        public decimal PriceWithTax
+        decimal tax;
+        decimal discount;
+        public decimal TotalPrice
         {
             get
             {
-                return BasicPrice + Tax.ProductTax(BasicPrice);
+                return BasicPrice + tax - discount;
             }
         }
-        public Product(string name, int upc, decimal priceWithPrice)
+        public Product(string name, int upc, decimal price)
         {
             Name = name;
             UPC = upc;
-            BasicPrice = priceWithPrice;
+            BasicPrice = price;
+            tax = PriceCalculations.ProductTax(BasicPrice);
+            discount = PriceCalculations.ProductDiscount(BasicPrice);
         }
```

Comment:
do you think placing the calculations in the constructor is a good choice? Is this its responsibility? 
the client/caller/user of this program should be able to call the calculations.
do you think whenever we change the value of the tax, or discount, we gonna make another product object?